cpp/src/decisiontree/memory.cuh

```diff
@@ -72,39 +73,63 @@ void TemporaryMemory<T, L>::LevelMemAllocator(int nrows, int ncols,
                                               int nbins, int depth,
                                               const int split_algo,
                                               bool col_shuffle) {
-  if (depth > 20 || (depth == -1)) {
-    max_nodes_per_level = pow(2, 20);
+  if (depth > swap_depth || (depth == -1)) {
+    max_nodes_per_level = pow(2, swap_depth);
   } else {
     max_nodes_per_level = pow(2, depth);
   }
-  int maxnodes = max_nodes_per_level;
-  int ncols_sampled = (int)(ncols * colper);
+  size_t maxnodes = max_nodes_per_level;
+  size_t ncols_sampled = (int)(ncols * colper);
+  size_t gather_max_nodes;
+  if (depth < 64) {
+    gather_max_nodes = std::min((size_t)(nrows + 1),
+                                (size_t)(pow((size_t)2, (size_t)depth) + 1));
+  } else {
+    gather_max_nodes = nrows + 1;
+  }
+  parentsz = std::max(maxnodes, gather_max_nodes);
+  childsz = std::max(2 * maxnodes, 2 * gather_max_nodes);
+
+  //std::cout << "maxnodes --> " << maxnodes << "  gather maxnodes--> "
+  //          << gather_max_nodes << std::endl;
+  //std::cout << "Parent size --> " << parentsz << std::endl;
+  //std::cout << "Child size  --> " << childsz << std::endl;
+  //std::cout << "Nrows size --> " << (nrows + 1) << std::endl;
+  //std::cout << "Sparse tree holder size --> " << 2 * gather_max_nodes
+  //          << std::endl;
```

Comment:
nitpick: remove commented code

Reply:
I would like to keep this, as it can be useful to debug in future. Is that a possibility ?

Reply:
then probably it is better to create a class method just to print all this info.

Reply:
Make sense. I will do that. Thanks